### PR TITLE
docs(manager/custom): hint about multiple replacements

### PR DESCRIPTION
**Pull request by Finkregh**
## Changes

No behaviour change, only documentation hint.

## Context

See e.g. https://github.com/renovatebot/renovate/discussions/20651

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

Only documentation change, no testing required IMHO.

### CODE REVIEW

File: lib/modules/manager/custom/regex/readme.md [hunk 1]
```diff
@@ -51,7 +51,7 @@ ENV YARN_VERSION=1.19.1
 RUN curl -o- -L https://yarnpkg.com/install.sh | bash -s -- --version ${YARN_VERSION}
 ```
 
-You would need to capture the `currentValue` with a named capture group, like this: `ENV YARN_VERSION=(?<currentValue>.*?)\\n`.
+You would need to capture the `currentValue` with a named capture group, like this: `ENV YARN_VERSION=(?<currentValue>.*?)\\n`. If you need to update a version string multiple times in a line you need to have multiple `matchStrings`, one for each occurence.
```
Comment (rarkins) [inline lib/modules/manager/custom/regex/readme.md:54]:
```suggestion
You would need to capture the `currentValue` with a named capture group, like this: `ENV YARN_VERSION=(?<currentValue>.*?)\\n`.
If you need to update a version string multiple times in a line you need to have multiple `matchStrings`, one for each occurence.
```

**Review by rarkins — CHANGES_REQUESTED** [review]

File: lib/modules/manager/custom/regex/readme.md [hunk 2]
```diff
@@ -51,7 +51,8 @@ ENV YARN_VERSION=1.19.1
 RUN curl -o- -L https://yarnpkg.com/install.sh | bash -s -- --version ${YARN_VERSION}
 ```
 
-You would need to capture the `currentValue` with a named capture group, like this: `ENV YARN_VERSION=(?<currentValue>.*?)\\n`. If you need to update a version string multiple times in a line you need to have multiple `matchStrings`, one for each occurence.
+You would need to capture the `currentValue` with a named capture group, like this: `ENV YARN_VERSION=(?<currentValue>.*?)\\n`.
+If you need to update a version string multiple times in a line you need to have multiple `matchStrings`, one for each occurence.
```
Comment (HonkingGoose) [inline lib/modules/manager/custom/regex/readme.md:55]:
```suggestion
To update a version string multiple times in a line: use multiple `matchStrings`, one for each occurrence.
```

**Review by HonkingGoose — CHANGES_REQUESTED** [review]

File: lib/modules/manager/custom/regex/readme.md [hunk 3]
```diff
@@ -52,6 +52,7 @@ RUN curl -o- -L https://yarnpkg.com/install.sh | bash -s -- --version ${YARN_VER
 ```
 
 You would need to capture the `currentValue` with a named capture group, like this: `ENV YARN_VERSION=(?<currentValue>.*?)\\n`.
+To update a version string multiple times in a line: use multiple `matchStrings`, one for each occurrence.
```
Comment (rarkins) [inline lib/modules/manager/custom/regex/readme.md:55]: Is it multiple matchStrings or multiple custom managers?

Comment (Finkregh) [reply lib/modules/manager/custom/regex/readme.md:55]: The configuration I am using:

```json5
  customManagers: [
    {
      customType: "regex",
      fileMatch: ["remote-images.txt"],
      matchStrings: [
        // for the version on the left part, ignoring the right
        "# renovate: datasource=(?<datasource>.*?) depName=(?<depName>.*?)( versioning=(?<versioning>.*?))?\\s\\S+?:(?<currentValue>\\S+)\\s+\\S+:.+",
        // for the version on the right part, ignoring the left
        "# renovate: datasource=(?<datasource>.*?) depName=(?<depName>.*?)( versioning=(?<versioning>.*?))?\\s\\S+?:\\S+\\s+\\S+:(?<currentValue>\\S+)",
      ],
      versioningTemplate: "{{#if versioning}}{{{versioning}}}{{else}}semver{{/if}}",
    },
  ],
```

For a file that looks like this:

```text
# renovate: datasource=github-tags depName=quay/clair versioning=loose
something:4.7.2    something-else:4.7.2
```

This updates both versions

Comment (rarkins) [reply lib/modules/manager/custom/regex/readme.md:55]: Could you add that as an example?

Comment (Finkregh) [reply lib/modules/manager/custom/regex/readme.md:56]: extended

File: lib/modules/manager/custom/regex/readme.md [hunk 4]
```diff
@@ -53,6 +53,35 @@ RUN curl -o- -L https://yarnpkg.com/install.sh | bash -s -- --version ${YARN_VER
 
 You would need to capture the `currentValue` with a named capture group, like this: `ENV YARN_VERSION=(?<currentValue>.*?)\\n`.
 
+To update a version string multiple times in a line: use multiple `matchStrings`, one for each occurrence.
+
+Here is the full renovate json5 config:
```
Comment (HonkingGoose) [inline lib/modules/manager/custom/regex/readme.md:58]:
```suggestion
Here is the full Renovate `.json5` config:
```

**Review by HonkingGoose — CHANGES_REQUESTED** [review]

**Review by rarkins — DISMISSED** [review]

**Review by rarkins — APPROVED** [review]

**Pull request merged**